instrumentation/hikaricp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/hikaricp/OpenTelemetryMetricsTracker.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.hikaricp;
+
+import com.zaxxer.hikari.metrics.IMetricsTracker;
+import com.zaxxer.hikari.metrics.PoolStats;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.ObservableLongUpDownCounter;
+import io.opentelemetry.instrumentation.api.metrics.db.DbConnectionPoolMetrics;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+final class OpenTelemetryMetricsTracker implements IMetricsTracker {
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.hikaricp-3.0";
+  private static final double NANOS_PER_MS = TimeUnit.MILLISECONDS.toNanos(1);
+
+  @Nullable private final IMetricsTracker userMetricsTracker;
+
+  private final List<ObservableLongUpDownCounter> observableInstruments;
+  private final LongCounter timeouts;
+  private final DoubleHistogram createTime;
+  private final DoubleHistogram waitTime;
+  private final DoubleHistogram useTime;
+  private final Attributes attributes;
+
+  public OpenTelemetryMetricsTracker(
+      @Nullable IMetricsTracker userMetricsTracker, String poolName, PoolStats poolStats) {
+
+    this.userMetricsTracker = userMetricsTracker;
+
+    DbConnectionPoolMetrics metrics =
+        DbConnectionPoolMetrics.create(GlobalOpenTelemetry.get(), INSTRUMENTATION_NAME, poolName);
+
+    this.observableInstruments =
+        Arrays.asList(
+            metrics.usedConnections(poolStats::getActiveConnections),
+            metrics.idleConnections(poolStats::getIdleConnections),
+            metrics.minIdleConnections(poolStats::getMinConnections),
+            metrics.maxConnections(poolStats::getMaxConnections),
+            metrics.pendingRequestsForConnection(poolStats::getPendingThreads));
+
+    this.timeouts = metrics.connectionTimeouts();
+    this.createTime = metrics.connectionCreateTime();
+    this.waitTime = metrics.connectionWaitTime();
+    this.useTime = metrics.connectionUseTime();
+    this.attributes = metrics.getAttributes();
+  }
```

Comment:
There's enough construction work happening inside this constructor that I think a builder could help. Or maybe the factory that creates this just isn't doing enough work.

Reply:
Yeah, I can move that to the factory class.